common/src/main/java/com/microsoft/identity/common/crypto/AndroidBrokerStorageEncryptionManager.java

```diff
@@ -0,0 +1,168 @@
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.crypto;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+
+import androidx.annotation.Nullable;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.adal.internal.AuthenticationSettings;
+import com.microsoft.identity.common.adal.internal.cache.StorageHelper;
+import com.microsoft.identity.common.java.crypto.StorageEncryptionManager;
+import com.microsoft.identity.common.java.crypto.key.AES256KeyLoader;
+import com.microsoft.identity.common.java.exception.ClientException;
+import com.microsoft.identity.common.java.telemetry.ITelemetryCallback;
+import com.microsoft.identity.common.logging.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import lombok.NonNull;
+
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Broker.AZURE_AUTHENTICATOR_APP_PACKAGE_NAME;
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Broker.BROKER_HOST_APP_PACKAGE_NAME;
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Broker.COMPANY_PORTAL_APP_PACKAGE_NAME;
+
+/**
+ * Key Encryption Manager for Broker.
+ */
+public class AndroidBrokerStorageEncryptionManager extends StorageEncryptionManager {
+    private static final String TAG = AndroidBrokerStorageEncryptionManager.class.getSimpleName();
+
+    /**
+     * A {@link SharedPreferences} key for storing decryption failure event.
+     * see {@link AndroidBrokerStorageEncryptionManager#handleDecryptionFailure}
+     */
+    private static final String CURRENT_ACTIVE_BROKER_SHAREDPREF_KEY = "current_active_broker";
+
+    protected final String LEGACY_AUTHENTICATOR_APP_KEY = "LEGACY_AUTHENTICATOR_APP_KEY";
+    protected final String LEGACY_COMPANY_PORTAL_KEY = "LEGACY_COMPANY_PORTAL_KEY";
```

Comment:
So...this is isn't the `key` but rather a constant to refer to the actual key.

perhaps call it `LEGACY_AUTHENTICATOR_APP_KEY_ALIAS` and same for CP